plugins/providers/hyperv/action/configure.rb

```diff
@@ -1,69 +1,128 @@
 require "fileutils"
-
 require "log4r"
+require "timeout"
+require "vagrant/util/network_ip"
 
 module VagrantPlugins
   module HyperV
     module Action
       class Configure
+
+        include Vagrant::Util::NetworkIP
+
         def initialize(app, env)
           @app = app
           @logger = Log4r::Logger.new("vagrant::hyperv::configure")
         end
 
         def call(env)
+          @env = env
+          env[:machine].provider.driver.execute(:install_managed_switches)
```

Comment:
At the moment I'm just adding 2 additional switches to hyper-v for each of the virtualbox switch types. `Default Switch` is a NAT and DHCP enabled special switch you can't delete that can be used in a similar fashion to the virtualbox NAT adapter.  
![image](https://user-images.githubusercontent.com/13159458/186911714-e67d8e76-eddc-4f01-bd3d-805de997b778.png)